File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {

        if (n == 1 || n == 2 || n == 3) {
            return n;
        }
        
        int pre = 1;
        int curr = 2;
        int res = 0;
        
        for (int i = 2; i < n; i++) {
            res = curr + pre;
            pre = curr;
            curr = res;
        }
        return res;
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.go
func climbStairs(n int) int {
    if n == 1 || n == 2 || n == 3 {
        return n
    }
    
    pre := 1
    curr := 2
    res := 0
    
    for i := 2; i < n; i++ {
        res = curr + pre
        pre = curr
        curr = res
    }
    return res
}